File: app/src/pfp/java/social/entourage/android/authentification/login/LoginPresenter.java
package social.entourage.android.authentification.login;


import javax.inject.Inject;

import social.entourage.android.api.LoginRequest;
import social.entourage.android.api.UserRequest;
import social.entourage.android.api.model.User;
import social.entourage.android.authentication.AuthenticationController;
import social.entourage.android.authentication.login.BaseLoginPresenter;
import social.entourage.android.authentication.login.LoginActivity;

/**
 * Presenter controlling the LoginActivity
 * @see LoginActivity
 */
public class LoginPresenter extends BaseLoginPresenter {

    @Inject
    public LoginPresenter(
            final LoginActivity activity,
            final LoginRequest loginRequest,
            final UserRequest userRequest,
            final AuthenticationController authenticationController) {
        super(activity, loginRequest, userRequest, authenticationController);
    }

    protected boolean shouldShowChoosePhotoAfterLogin() {
        User me = authenticationController.getUser();
        return (me != null && (me.getAvatarURL() == null || me.getAvatarURL().length() == 0));
    }
}

File: app/src/pfp/java/social/entourage/android/authentification/login/LoginPresenter.kt
package social.entourage.android.authentification.login

import social.entourage.android.api.LoginRequest
import social.entourage.android.api.UserRequest
import social.entourage.android.api.model.User
import social.entourage.android.authentication.AuthenticationController
import social.entourage.android.authentication.login.BaseLoginPresenter
import social.entourage.android.authentication.login.LoginActivity
import javax.inject.Inject

/**
 * Presenter controlling the LoginActivity
 * @see LoginActivity
 */
class LoginPresenter @Inject constructor(
        activity: LoginActivity?,
        loginRequest: LoginRequest?,
        userRequest: UserRequest?,
        authenticationController: AuthenticationController?) : BaseLoginPresenter(activity, loginRequest, userRequest, authenticationController) {
    protected fun shouldShowChoosePhotoAfterLogin(): Boolean {
        val me = authenticationController.user
        return me != null && (me.avatarURL == null || me.avatarURL.length == 0)
    }

    override fun shouldShowTermsAndConditions(): Boolean {
        return true
    }

    override fun shouldContinueWithRegistration(): Boolean {
        return false
    }

    override fun shouldShowNameView(user: User): Boolean {
        return super.shouldShowNameView(user) || authenticationController.isNewUser
    }

    override fun shouldShowEmailView(user: User): Boolean {
        return super.shouldShowEmailView(user) || authenticationController.isNewUser
    }
}